src/colmap/scene/image.h

```diff
@@ -164,9 +185,46 @@ point2D_t Image::NumPoints2D() const {
 
 point2D_t Image::NumPoints3D() const { return num_points3D_; }
 
-const Rigid3d& Image::CamFromWorld() const { return cam_from_world_; }
+frame_t Image::FrameId() const { return frame_id_; }
+
+void Image::SetFrameId(frame_t frame_id) { frame_id_ = frame_id; }
+
+const std::shared_ptr<class Frame> Image::Frame() const { return frame_; }
+
+void Image::SetFrame(std::shared_ptr<class Frame> frame) {
+  frame_ = std::move(frame);
+}
+
+bool Image::HasFrame() const { return frame_ != nullptr; }
+
+bool Image::HasNonTrivialFrame() const {
+  return HasFrame() && frame_->HasRigCalibration() &&
+         !frame_->RigCalibration()->IsReference(
```

Comment:
This assumes that one of the cameras always coincides with the rigs origin? This may not be a convenient choice in case when there is an IMU, where it might be easiest to model everything in the IMU frame?

Reply:
There might be some misunderstanding here. We can set the IMU to be the reference frame in the current ``Frame`` class to support this?

Reply:
I see, got it. I read through the code top to bottom and assumed the "Frame" was only used for cameras. This makes sense. The only scenario that is not possible is to have an origin that doesn't coincide with any of the sensors. Not sure if this is useful but I currently also don't see why we would need to have one of them assigned as the reference? Maybe you can explain your motivation here. Thanks.

Reply:
Thanks. It could make sense to add this motivation somewhere in the header?